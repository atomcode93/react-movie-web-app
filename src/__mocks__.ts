export const initialState = {
  auth: {
    error: false,
    isAuthenticated: false,
    user: null,
  },
  favorites: {
    error: false,
    favoritedIds: [],
    favoritedMovies: null,
  },
  filters: {
    order: {
      label: 'Descending',
      value: 'desc',
    },
    rating: {
      max: 10,
      min: 5,
    },
    runtime: {
      max: 250,
      min: 45,
    },
    sort_by: {
      label: 'Rating',
      value: 'vote_average',
    },
    year: 2019,
  },
  modals: {
    args: null,
    open: false,
    type: null,
  },
  movies: {
    error: false,
    favorites: [],
    loading: false,
    movie: null,
    movies: null,
    popularMovies: null,
    soonMovies: null,
    topRatedMovies: null,
    watchLater: null,
  },
  router: {
    action: 'POP',
    location: {
      hash: '',
      pathname: '/',
      search: '',
      state: undefined,
    },
  },
  watchLater: {
    error: false,
    watchLaterIds: [],
    watchLaterMovies: null,
  },
};
