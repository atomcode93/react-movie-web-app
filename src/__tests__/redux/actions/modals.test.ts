import { actions as modalsActions } from 'redux/ducks/modals.duck';
import { initialState } from '__mocks__';

describe('Filters Actions', () => {
  test('should open modal', () => {
    const action = modalsActions.openModal({
      type: 'video',
    });
    expect(action).toEqual({
      type: 'OPEN_MODAL',
      payload: {
        type: 'video',
      },
    });
  });
  test('should close modal', () => {
    const action = modalsActions.closeModal();
    expect(action).toMatchSnapshot();
  });
});
