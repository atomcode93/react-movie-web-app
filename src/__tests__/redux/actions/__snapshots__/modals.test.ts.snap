// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters Actions should close modal 1`] = `
Object {
  "type": "CLOSE_MODAL",
}
`;
